lib/descripting/cog.cpp

```diff
@@ -10,12 +10,23 @@
 
 namespace MouseTrack {
 
-/// Takes a cluster and its referencing PointCloud and creates a cluster descriptor
-std::unique_ptr<ClusterDescriptor> CenterOfGravity::operator()(const Cluster& cluster, const PointCloud& cloud) const {
-	std::unique_ptr<ClusterDescriptor> descriptor(new Center());
-	return descriptor;
+/// Takes a cluster and its referencing PointCloud and creates a cluster
+/// descriptor
+std::unique_ptr<ClusterDescriptor> CenterOfGravity::operator()(
+    const Cluster &cluster, const PointCloud &cloud) const {
+  Coordinate sumX = 0, sumY = 0, sumZ = 0;
+  size_t numPoints = cluster.points().size();
+  // Go through each point in the cluster
+  for (PointIndex i = 0; i < cluster.points().size(); i++) {
+    // Get the sum of all coordinates along x, y ,z
+    sumX += cloud[i].x();
+    sumY += cloud[i].y();
+    sumZ += cloud[i].z();
+  }
```

Comment:
For optimization phase: we can use Eigen::Vector3d here.

Reply:
I would definitely not change it at the moment. Clang (and other compilers) might be able to vectorize this loop. In that case it might be more efficient than creating Eigen vectors.